create v6 upgrade guide page (#12480)
withastro/docs: src/content/docs/en/guides/upgrade-to/v6.mdx

<ReadMore>Learn more about [prefetching](/en/guides/prefetch/).</ReadMore>

> ### Experimental Flags

Reviewer: @florian-lefebvre I added this section (copied from v5 guide) so that we could show removing experimental flags, too. Then below, we can mention this section as where to go to see how to remove (all) flags, rather than add more to the listing itself.

The following experimental flags have been removed in Astro v5.0 and these features are available for use:

- `thing 1`

Additionally, the following experimental flags have been removed and **are now the default or recommended behavior in Astro v5.0**.

- `experimental.preserveScriptOrder` (See below for breaking changes to [default `<script>` and `<style>` behavior](#changed-script-and-style-tags-are-rendered-in-the-order-they-are-defined).)

> The following experimental flags have been removed and **their corresponding features are not part of Astro v5.0**.

Reviewer: ```suggestion
The following experimental flags have been removed and **their corresponding features are not part of Astro v6.0**.
```

- `thing 2`

> Remove these experimental flags if you were previously using them, and move your `env` configuration to the root of your Astro config:

Reviewer: ```suggestion
Remove these experimental flags from your Astro config if you were previously using them:
```

<ReadMore>Learn more about [Internationalization routing](/en/guides/internationalization/#routing).</ReadMore>

### Changed: `<script>` and `<style>` tags are rendered in the order they are defined

<SourcePR number="14480" title="feat: stabilize experimental preserveScriptOrder option"/>

In Astro v5.5, the `experimental.preserveScriptOrder` flag was introduced to render multiple `<style>` and `<script>` tags in the same order as they were declared in the source code. Astro 5.x reversed their order in your generated HTML output. This could give unexpected results, for example, CSS styles being overridden by earlier defined style tags when your site was built.

Astro 6.0 removes this experimental flag and makes this the new default behavior in Astro: scripts and styles are now rendered in the order defined in your code.

#### What should I do?

> If you were previously using this experimental feature, you must [remove this experimental flag from your configuration](#experimental-flags) as it no longer exists.

Reviewer: We can use this as a standard line for all of these!
